confidential-data-hub/storage/src/error.rs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2023 Intel
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use thiserror::Error;
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("secure mount failed: {0}")]
+    SecureMountFailed(String),
```

Comment:
Nice to see that `thiserror` crate is used!

I suggest that we have some more fine grained error types. In fact `SecureMountFailed` applies all the errors returned from the crate.

The `storage` is somehow similiar with `kms`. I divided the errors by the plugin type here https://github.com/confidential-containers/guest-components/blob/main/confidential-data-hub/kms/src/error.rs  and I know that is not perfect. Please think about this and refer to this in your preferred way

Reply:
do some changes